modules/ROOT/pages/deriving-container.adoc

```diff
@@ -0,0 +1,127 @@
+= Deriving from Fedora CoreOS as a bootable container image
+
+NOTICE: At the current time, the suggested model here does not integrate with the current "update barriers" model used for default Fedora CoreOS updates.  While relatively rare so far, this means that some container-based updates may see issues that were otherwise gated.  This issue is being worked on.  More information in https://github.com/coreos/fedora-coreos-tracker/issues/1263[this issue].
+
+== Bootable containers overview
+
+The operating system images used for Fedora CoreOS are available as a standard container image; there is one per stream. See xref:update-streams.adoc[Update Streams].
+
+- quay.io/fedora/fedora-coreos:stable
+- quay.io/fedora/fedora-coreos:next
+- quay.io/fedora/fedora-coreos:testing
+
+== Creating custom builds
+
+See the https://github.com/coreos/layering-examples[layering examples] git repository.  While the examples there tend to use Containerfile/Dockerfile style builds, you can use any tool which can build containers and push the result to a registry.
+
+The default suggested approach is to have container tags be the source of truth for initating upgrades, and to have the client systems poll.  Example systemd units for this are in [https://github.com/coreos/layering-examples/tree/main/autoupdate-host]; they boil down to a systemd `.service` which just runs `rpm-ostree upgrade --reboot` and a corresponding `.timer` unit to run it once a day.
+
+Of course, client side logic could be much more complex, including a privileged container image that runs completely arbitrary logic.
+
+== Using Ignition to switch to a bootable container image
+
+The https://coreos.github.io/rpm-ostree/container/[rpm-ostree container page] describes the commands for interacting with bootable ostree container images.
+
+This systemd unit will run only on the first boot, and switch a "stock" Fedora CoreOS node to your custom image.
+
+[source,yaml]
+----
+variant: fcos
+version: 1.4.0
+systemd:
+  units:
+    # Our custom unit for rebasing
+    - name: rebase-custom.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Fetch and deploy target image
+        # Only run on the firstboot
+        ConditionFirstBoot=true
+        After=network-online.target
+        [Service]
+        # This ordering is important
+        After=ignition-firstboot-complete.service
```

Comment:
I think this is now `coreos-ignition-firstboot-complete.service`

Reply:
![image](https://github.com/coreos/fedora-coreos-docs/assets/1939288/0dcff2b2-ea41-461d-a9f9-d4d6f71a5b1f)

From 

```
[core@forgejo-actionsrunner ~]$ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="39.20240210.3.0 (CoreOS)"
ID=fedora
VERSION_ID=39
```